ci-operator/step-registry/openstack/provision/mirror/openstack-provision-mirror-commands.sh

```diff
@@ -0,0 +1,141 @@
+mirror_ipv4="$(openstack server show -f value -c addresses mirror-$CLUSTER_NAME-${CONFIG_TYPE} | grep -oE $IPV4_REGEX)"
+mirror_ipv6="$(openstack server show -f value -c addresses mirror-$CLUSTER_NAME-${CONFIG_TYPE} | grep -oE $IPV6_REGEX)"
+
+# configure the local container environment to have the correct SSH configuration
+if ! whoami &> /dev/null; then
+    if [[ -w /etc/passwd ]]; then
+        echo "${BASTION_USER}:x:$(id -u):0:${BASTION_USER} user:${HOME}:/sbin/nologin" >> /etc/passwd
+    fi
+fi
+
+if [[ -f "${SHARED_DIR}/squid-credentials.txt" ]]; then
+   proxy_host=$(yq -r ".clouds.${OS_CLOUD}.auth.auth_url" "$OS_CLIENT_CONFIG_FILE" | cut -d/ -f3 | cut -d: -f1)
+   proxy_credentials=$(<"${SHARED_DIR}/squid-credentials.txt")
+   echo "Using proxy $proxy_host for SSH connection"
+else
+  echo "Missing squid-credentials.txt file"
+  exit 1
+fi
+
+ssh_via_proxy() {
+    local command=$1
+    ssh -o ConnectTimeout=10 -o StrictHostKeyChecking=no -i ${CLUSTER_PROFILE_DIR}/ssh-privatekey -o ProxyCommand="nc --proxy-auth ${proxy_credentials} --proxy ${proxy_host}:3128 %h %p" $BASTION_USER@$mirror_ipv4 "$command"
+}
+
+scp_via_proxy() {
+    local src=$1
+    local dest=$2
+    scp -o ConnectTimeout=10 -o StrictHostKeyChecking=no -i ${CLUSTER_PROFILE_DIR}/ssh-privatekey -o ProxyCommand="nc --proxy-auth ${proxy_credentials} --proxy ${proxy_host}:3128 %h %p" $src $dest
+}
+
+if ! retry 60 5 ssh_via_proxy "uname -a"; then
+		echo "ERROR: Bastion proxy is not reachable via $mirror_ipv4 - check logs:"
+    openstack console log show ${server_id}
+		exit 1
+fi
+
+MIRROR_REGISTRY_DNS_NAME="mirror-registry.${CLUSTER_NAME}.${BASE_DOMAIN}"
+MIRROR_REGISTRY_CREDENTIALS=$(<"/var/run/vault/mirror-registry/registry_creds")
+scp_via_proxy "/var/run/vault/mirror-registry/registry_creds_encrypted_htpasswd" $BASTION_USER@$mirror_ipv4:/tmp/htpasswd
+
+echo "Deploying the mirror registry"
+>&2 cat << EOF > $WORK_DIR/deploy_mirror.sh
+#!/usr/bin/env bash
+set -e
+sudo mkdir -p /opt/registry/{auth,certs,data}
+sudo openssl req -newkey rsa:4096 -nodes -sha256 -keyout /opt/registry/certs/domain.key -x509 -days 1 -subj "/CN=mirror-$CLUSTER_NAME-${CONFIG_TYPE}" -addext "subjectAltName=DNS:$MIRROR_REGISTRY_DNS_NAME,DNS:mirror-$CLUSTER_NAME-${CONFIG_TYPE}" -out /opt/registry/certs/domain.crt
+sudo cp /opt/registry/certs/domain.crt /etc/pki/ca-trust/source/anchors/domain.crt   
+sudo mv /tmp/htpasswd /opt/registry/auth/htpasswd
+sudo update-ca-trust
+sudo dnf install -y podman
+sudo podman create --name registry -p 5000:5000 --net host \
```

Comment:
There is a tool available to create a registry https://docs.openshift.com/container-platform/4.16/installing/disconnected_install/installing-mirroring-creating-registry.html#mirror-registry-localhost_installing-mirroring-creating-registry

we might want to take a look at that as second phase.

Reply:
nice, I just found that it's being used in some steps already:

```
sudo yum install podman openssl -y
curl -L -o mirror-registry.tar.gz https://developers.redhat.com/content-gateway/rest/mirror/pub/openshift-v4/clients/mirror-registry/latest/mirror-registry.tar.gz
tar -xzvf mirror-registry.tar.gz
sudo ./mirror-registry install --quayHostname ${OMR_HOST_NAME} --initPassword password --initUser quay -v
```

I'll look at it in a second iteration.